public_asset/lib/data/network/public_asset_api.dart

```diff
@@ -0,0 +1,124 @@
+        .toJson(publicAsset.identityIds!),
+    });
+
+    return MapEntry(publicAsset.id!, patchObject);
+  }
+
+  Future<List<PublicAsset>> getPublicAssets(
+    Session session,
+    AccountId accountId,
+    {required List<Id> publicAssetIds}
+  ) async {
+    _checkForSupportedCapability(session);
+    final requestBuilder = JmapRequestBuilder(_httpClient, ProcessingInvocation());
+    final method = GetPublicAssetMethod(accountId);
+    method.addIds(publicAssetIds.toSet());
+    final invocation = requestBuilder.invocation(method);
+    final response = await (requestBuilder..usings(method.requiredCapabilities))
+      .build()
+      .execute();
+    final publicAssets = response.parse<GetPublicAssetResponse>(
+        invocation.methodCallId,
+        GetPublicAssetResponse.deserialize
+    )?.list ?? [];
+
+    return publicAssets;
+  }
+
+  Future<PublicAsset> createPublicAsset(
+    Session session,
+    AccountId accountId,
+    {
+      required Id blobId,
+      required IdentityId identityId
+    }
+  ) async {
+    _checkForSupportedCapability(session);
+    final requestBuilder = JmapRequestBuilder(_httpClient, ProcessingInvocation());
+    final method = SetPublicAssetMethod(accountId);
+    final generateCreateId = Id(_uuid.v1());
+    method.addCreate(
+      generateCreateId,
+      PublicAsset(blobId: blobId, identityIds: {identityId: true}));
+    final invocation = requestBuilder.invocation(method);
+    final response = await (requestBuilder..usings(method.requiredCapabilities))
+      .build()
+      .execute();
+    final publicAsset = response.parse<SetPublicAssetResponse>(
+        invocation.methodCallId,
+        SetPublicAssetResponse.deserialize
+    )?.created?[generateCreateId];
+
+    if (publicAsset == null) {
+      throw const CannotCreatePublicAssetException();
+    }
+
+    return publicAsset;
+  }
+
+  Future<void> destroyPublicAssets(
+    Session session,
+    AccountId accountId,
+    {required List<Id> publicAssetIds}
+  ) async {
+    _checkForSupportedCapability(session);
+    final requestBuilder = JmapRequestBuilder(_httpClient, ProcessingInvocation());
+    final method = SetPublicAssetMethod(accountId);
+    method.addDestroy(publicAssetIds.toSet());
+    await (requestBuilder..usings(method.requiredCapabilities))
+      .build()
+      .execute();
+  }
+
+  Future<void> updatePublicAssets(
+    Session session,
+    AccountId accountId,
+    {required List<PublicAsset> publicAssets}
+  ) async {
+    _checkForSupportedCapability(session);
+    final requestBuilder = JmapRequestBuilder(_httpClient, ProcessingInvocation());
+    final method = SetPublicAssetMethod(accountId);
+    method.addUpdates(
+      Map.fromEntries(
+        publicAssets
+          .where((publicAsset) => publicAsset.id != null && publicAsset.identityIds != null)
+          .map(_toPatchObjectMapEntry)
+      )
+    );
+    await (requestBuilder..usings(method.requiredCapabilities))
+      .build()
+      .execute();
```

Comment:
idem

Reply:
We are not concerning if the request is success or not, identity is still updated regardless of the result.

Reply:
the same with destroy

Reply:
Done